src/server.ts

```diff
@@ -251,14 +250,9 @@ export async function startPluginsServer(
         pubSub.on('message', async (channel: string, message) => {
             if (channel === server!.PLUGINS_RELOAD_PUBSUB_CHANNEL) {
                 status.info('⚡', 'Reloading plugins!')
-                await queue?.stop()
-                await stopSchedule?.()
-                if (piscina) {
-                    await stopPiscina(piscina)
-                }
-                piscina = makePiscina(serverConfig!)
-                queue = await startQueue(server!, piscina)
-                stopSchedule = await startSchedule(server!, piscina)
+
+                await piscina?.broadcastTask({ task: 'reloadPlugins' })
+                await scheduleControl?.reloadSchedule()
```

Comment:
I wonder what would happen if this gets called in the middle or beginning of a Kafka batch? `pluginConfig.vm` will be undefined for a short while on some plugins, possibly resulting in some crashes? Some logic to reload only after the batch has finished... and/or to make sure there's always some object on `vm` might make sense?

Reply:
You mean marius' comment? No.

This line? Absolutely needed. :)

Reply:
Yep, meant the comment!